### PR TITLE
New Package: xM4ddy.OFGB version 0.1

### PR DESCRIPTION
**Pull request by og-mrk**
Checklist for Pull Requests
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Is there a linked Issue?

Manifests
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] This PR only modifies one (1) manifest
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/doc/Authoring.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.6 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.6.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

---

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/157395)

### CODE REVIEW

File: manifests/x/xM4ddy/OFGB/0.1/xM4ddy.OFGB.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.6.0.schema.json
+
+PackageIdentifier: xM4ddy.OFGB
+PackageVersion: "0.1"
+InstallerType: portable
+ReleaseDate: 2024-05-01
+Dependencies:
+  PackageDependencies:
+    - PackageIdentifier: Microsoft.DotNet.Runtime.8
+Installers:
```
Comment (stephengillie) [inline manifests/x/xM4ddy/OFGB/0.1/xM4ddy.OFGB.installer.yaml:11]:
```suggestion
Dependencies:
  PackageDependencies:
    - PackageIdentifier: Microsoft.DotNet.DesktopRuntime.8
Installers:
```

**Review by stephengillie — CHANGES_REQUESTED** [review]
Manual Validation ended with:
image
image

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 
> Faulting application name: OFGB.exe, version: 1.0.0.0, time stamp: 0x65f90000 
> Faulting module name: KERNELBASE.dll, version: 10.0.19041.1202, time stamp: 0xc9db1934 Exception code: 0xe0434352 Fault offset: 0x0000000000034f99 
> Faulting process id: 0x1d98 
> Faulting application start time: 0x01dabcd6ee8f10ca 
> Faulting application path: C:\Users\User\AppData\Local\Microsoft\WinGet\Links\OFGB.exe 
> Faulting module path: C:\Windows\System32\KERNELBASE.dll Report Id: afe2b2ff-f6e9-4d13-b7f3-8e005b318922 
> Faulting package full name:  
> Faulting package-relative application ID:    Application: OFGB.exe CoreCLR Version: 8.0.624.26715 .NET Version: 8.0.6 Description: The process was terminated due to an unhandled exception. Exception Info: System.Windows.Markup.XamlParseException: The invocation of the constructor on type 'OFGB.MainWindow' that matches the specified binding constraints threw an exception.  ---> System.InvalidOperationException: Error Initializing While Creating Key    at OFGB.MainWindow.CreateKey(String loc, String key)    at OFGB.MainWindow.InitializeKeys()    at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)    --- End of inner exception stack trace ---    at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)    at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)    at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)    at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)    at System.Windows.Application.LoadBamlStreamWithSyncInfo(Stream stream, ParserContext pc)    at System.Windows.Application.DoStartup()    at System.Windows.Application.<.ctor>b__1_0(Object unused)    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)    at System.Windows.Threading.DispatcherOperation.InvokeImpl()    at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) --- End of stack trace from previous location ---    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)    at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)    at System.Windows.Threading.DispatcherOperation.Invoke()    at System.Windows.Threading.Dispatcher.ProcessQueue()    at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)    at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)    at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)    at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)    at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)    at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)    at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)    at System.Windows.Application.RunDispatcher(Object ignore)    at System.Windows.Application.RunInternal(Window window)    at OFGB.App.Main() 

(Automated response - build 885.)

**Comment by stephengillie** [comment]
Manual Validation ended with:
> 10:36:29 AM: 
> Faulting application name: OFGB-Deps.exe, version: 1.0.0.0, time stamp: 0x65f9e99a 
> Faulting module name: KERNELBASE.dll, version: 10.0.22621.3733, time stamp: 0x44653e19 Exception code: 0xe0434352 Fault offset: 0x000000000005f39c 
> Faulting process id: 0x0xd44 
> Faulting application start time: 0x0x1dabe813a8f2505 
> Faulting application path: C:\Users\User\AppData\Local\Microsoft\WinGet\Packages\xM4ddy.OFGB__DefaultSource\OFGB-Deps.exe 
> Faulting module path: C:\Windows\System32\KERNELBASE.dll Report Id: 54d9ec8d-b468-48db-85eb-fc41b871ee9a 
> Faulting package full name: 
> Faulting package-relative application ID: 10:36:29 AM: Application: OFGB-Deps.exe CoreCLR Version: 8.0.424.16909 .NET Version: 8.0.4 Description: The process was terminated due to an unhandled exception. Exception Info: System.Windows.Markup.XamlParseException: The invocation of the constructor on type 'OFGB.MainWindow' that matches the specified binding constraints threw an exception.  ---> System.InvalidOperationException: Error Initializing While Creating Key    at OFGB.MainWindow.CreateKey(String loc, String key)    at OFGB.MainWindow.InitializeKeys()    at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)    --- End of inner exception stack trace ---    at System.Windows.Markup.XamlReader.RewrapException(Exception e, IXamlLineInfo lineInfo, Uri baseUri)    at System.Windows.Markup.WpfXamlLoader.Load(XamlReader xamlReader, IXamlObjectWriterFactory writerFactory, Boolean skipJournaledProperties, Object rootObject, XamlObjectWriterSettings settings, Uri baseUri)    at System.Windows.Markup.WpfXamlLoader.LoadBaml(XamlReader xamlReader, Boolean skipJournaledProperties, Object rootObject, XamlAccessLevel accessLevel, Uri baseUri)    at System.Windows.Markup.XamlReader.LoadBaml(Stream stream, ParserContext parserContext, Object parent, Boolean closeStream)    at System.Windows.Application.LoadBamlStreamWithSyncInfo(Stream stream, ParserContext pc)    at System.Windows.Application.DoStartup()    at System.Windows.Application.<.ctor>b__1_0(Object unused)    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)    at System.Windows.Threading.DispatcherOperation.InvokeImpl()    at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state) --- End of stack trace from previous location ---    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)    at MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)    at System.Windows.Threading.DispatcherOperation.Invoke()    at System.Windows.Threading.Dispatcher.ProcessQueue()    at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)    at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)    at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)    at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)    at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)    at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)    at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)    at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)    at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)    at System.Windows.Application.RunDispatcher(Object ignore)    at System.Windows.Application.RunInternal(Window window)    at OFGB.App.Main()

(Automated response - build 885.)

**Comment by og-mrk** [comment]
> This application installs normally on a Windows 10 VM. But it doesn't seem to have any output when run.
> 
> > ShortDescription: GUI Tool To Removes Ads From Various Places Around **Windows 11**
> 
> Does this manifest need `MinimumOSVersion: 10.0.22621.0`?

Hello @stephengillie
I haven't thought about this before, so the tests were failing for windows 10 for it... as this App expects Windows 11 to be installed on the system, just like you've pointed out.

Thanks for all the help to getting this manifest working correctly, many appreciations Stephen 👍

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Signature Update Failed.
> Failure to parse win 10, attempting win 11
> Installation failed with exit code -1978335216

(Automated response - build 886.)

**Comment by og-mrk** [comment]
> Automatic Validation ended with:
> 
> > Signature Update Failed.
> > Failure to parse win 10, attempting win 11
> > Installation failed with exit code -1978335216
> 
> (Automated response - build 886.)

Looks like the Test pipeline Fails upon not finding an installer to use, even though the Installer Manifest has explicitly stated that all installer entries require windows 11 (22H2) or above to continue through the installation process, meaning for this manifest, there's no windows 10 compatible installer to use, which's the desired behavior as this app is intended to be used in a Windows 11 environment.

If there's any other problems in the manifest, I'll try fixing them as soon as possible, thanks in advance 😄

**Comment by og-mrk** [comment]
Does the testing pipeline expects all packages to work on both Windows 10 & 11? @stephengillie 
If so.. then this app can not be added to the WinGet Packages Repo.

**Comment by stephengillie** [comment]
HI @og-mrk,

I do not know why this package isn't triggering a Windows 11 Pipeline VM. For manual validation, I need to set up a Windows 11 VM again.

**Comment by og-mrk** [comment]
> HI @og-mrk,
> 
> I do not know why this package isn't triggering a Windows 11 Pipeline VM. For manual validation, I need to set up a Windows 11 VM again.

Will make a new PR based on this, which'll add the latest version of this app (version 0.4 as of time of writing), to see if the problem lies on the Application side instead of testing pipeline (as this's the first version released by the author).

**Comment by stephengillie** [comment]
Close with reason: Duplicate of #160130;

**Comment by stephengillie** [comment]
Seeing this is an earlier version, not a duplicate.

**Review by stephengillie — APPROVED** [review]

**Pull request merged**